### ISSUE TITLE
Having a mistake in room script prevents from saving the room itself.

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#1627]
**Describe the bug**
If room script compilation fails, the room cannot be not saved.

The root of the problem is that in AGS 3.* rooms are binary "blobs" which contain both room data and compiled scripts. So theoretically room data + compiled script supposed to be in sync. On another hand, the script (source) itself is a separate *.asc file.

This situation may be quite frustrating for the user, as they may not want or have time to fix the script, but would like to save the modified room. Being unable to do so will force them to loose the recent changes.

I think it may be done so that the room is saved with the older compiled script version if compiling new fails, provided there's a clear warning for the user.

NOTE: also has to double check what is a behavior in a same situation with the open room format in ags4 branch.

**AGS Version**
3.6.0, and any earlier.

**To Reproduce**
1. Create a room.
2. Write some bad syntax in the room's script.
3. Do changes to the room itself (create new objects, edit masks, and so on)
4. Try to save the room.
5. Observe the problem.

**Expected behavior**
IMO in case of compilation error, room data should be saved with the previously existing compiled script, and a warning is displayed for the user.

**Comment by ericoporto** [comment]
I think AGS can already load room script object (the result of the compilation of the script) separate from the rest of the room object, since #1330.

**Comment by ivan-mogilko** [comment]
Yes, one alternative is to stop saving compiled scripts in crm file, and generate .o files to save along with crm. That would require some changes to game building process, and perhaps a place for intermediate files.
